RecoBTag/SecondaryVertex/src/GhostTrackComputer.cc

```diff
@@ -316,3 +353,220 @@ GhostTrackComputer::operator () (const TrackIPTagInfo &ipInfo,
 
 	return vars;
 }
+
+TaggingVariableList
+GhostTrackComputer::operator () (const CandIPTagInfo &ipInfo,
+                                 const CandSecondaryVertexTagInfo &svInfo) const
+{
+	using namespace ROOT::Math;
+
+	edm::RefToBase<Jet> jet = ipInfo.jet();
+	math::XYZVector jetDir = jet->momentum().Unit();
+	bool havePv = ipInfo.primaryVertex().isNonnull();
+	GlobalPoint pv;
+	if (havePv)
+		pv = GlobalPoint(ipInfo.primaryVertex()->x(),
+		                 ipInfo.primaryVertex()->y(),
+		                 ipInfo.primaryVertex()->z());
+
+	btag::Vertices::VertexType vtxType = btag::Vertices::NoVertex;
+
+	TaggingVariableList vars;
+
+	vars.insert(btau::jetPt, jet->pt(), true);
+	vars.insert(btau::jetEta, jet->eta(), true);
+
+	TrackKinematics allKinematics;
+	TrackKinematics vertexKinematics;
+	TrackKinematics trackKinematics;
+
+	std::pair<double, double> vertexDist2D, vertexDist3D;
+	std::pair<double, double> tracksDist2D, tracksDist3D;
+
+	unsigned int nVertices = 0;
+	unsigned int nVertexTracks = 0;
+	unsigned int nTracks = 0;
+	for(unsigned int i = 0; i < svInfo.nVertices(); i++) {
+//		const Vertex &vertex = svInfo.secondaryVertex(i);
+//		bool hasRefittedTracks = vertex.hasRefittedTracks();
+//		TrackRefVector tracks = svInfo.vertexTracks(i);
+		const reco::VertexCompositePtrCandidate &vertex = svInfo.secondaryVertex(i);
+		const std::vector<CandidatePtr> & tracks = vertex.daughterPtrVector();
+		unsigned int n = 0;
+		for(std::vector<CandidatePtr>::const_iterator track = tracks.begin(); track != tracks.end(); ++track)
+//			if (svInfo.trackWeight(i, *track) >= minTrackWeight) 
+				n++;
+
+		if (n < 1)
+			continue;
+		bool isTrackVertex = (n == 1);
+		++*(isTrackVertex ? &nTracks : &nVertices);
+
+		addMeas(*(isTrackVertex ? &tracksDist2D : &vertexDist2D),
+					svInfo.flightDistance(i, true));
+		addMeas(*(isTrackVertex ? &tracksDist3D : &vertexDist3D),
+					svInfo.flightDistance(i, false));
+
+		TrackKinematics &kin = isTrackVertex ? trackKinematics : vertexKinematics;
+
+		for(std::vector<CandidatePtr>::const_iterator track = tracks.begin(); track != tracks.end(); ++track) {
+			kin.add(*(*track)->bestTrack(), 1.0);
+//		for(TrackRefVector::const_iterator track = tracks.begin(); track != tracks.end(); track++) {
```

Comment:
here and in other files, please remove commented out code, especially if it's a remainder of now unnecessary debugging step.
If it is needed for some reason, add some comments why, or even compile it and enable with relevant configuration or even compilation switches (ifdef)